tests/conftest.py

```diff
@@ -622,18 +624,39 @@ def xmlrpc(self, path, method, *args):
 
 
 @pytest.fixture
-def webtest(app_config):
-    # TODO: Ensure that we have per test isolation of the database level
-    #       changes. This probably involves flushing the database or something
-    #       between test cases to wipe any committed changes.
+def tm():
+    tm = transaction.TransactionManager(explicit=True)
+    tm.begin()
+    tm.doom()
 
+    yield tm
+
+    tm.abort()
+
+
+@pytest.fixture
+def webtest(app_config, tm, db_session):
     # We want to disable anything that relies on TLS here.
     app_config.add_settings(enforce_https=False)
 
-    try:
-        yield _TestApp(app_config.make_wsgi_app())
-    finally:
-        app_config.registry["sqlalchemy.engine"].dispose()
+    # Register the transaction manager with the db_session
+    zope.sqlalchemy.register(db_session, transaction_manager=tm)
```

Comment:
I _think_ this can be removed safely, since the tm is being set explicitly later on in `testapp`

Reply:
I didn't find out what magic eventually registers it, but seems likely that it should happen since we define `tm.manager`. I've removed it and it continued to work... :crossed_fingers: